catalog/app/containers/Bucket/PackageList/PackageList.tsx

```diff
@@ -317,49 +350,48 @@ export default function PackageList({
     params: { bucket },
   },
   location,
-}) {
-  const history = useHistory()
-  const s3 = AWS.S3.use()
-  const req = APIConnector.use()
-  // const sign = AWS.Signer.useS3Signer()
-  // const { analyticsBucket, apiGatewayEndpoint: endpoint } = Config.useConfig()
-  const { analyticsBucket } = Config.useConfig()
+}: RRDom.RouteComponentProps<{ bucket: string }>) {
+  const history = RRDom.useHistory()
   const { urls } = NamedRoutes.use()
-  // const bucketCfg = BucketConfig.useCurrentBucketConfig()
   const classes = useStyles()
 
-  const scrollRef = React.useRef(null)
+  const scrollRef = React.useRef<HTMLDivElement | null>(null)
 
-  const { sort, filter, p } = parseSearch(location.search)
-  const page = p && parseInt(p, 10)
+  const { sort, filter, p } = parseSearch(location.search, true)
+  const page = p ? parseInt(p, 10) : undefined
```

Comment:
> Can we use null here?

here -- yes, but not everywhere (e.g. not as a `title` attribute for a `span`, since it only accepts `string | undefined`)

> I think that it's more straightforward to use null, and leave undefined only when it's really undefined

tbh i don't feel much difference in most cases 🤷 (with a notable exception of `JSON.strignify`)